api/products.thrift

```diff
@@ -0,0 +1,78 @@
+// -------------------------------------------------------------------------
+//                     The CodeChecker Infrastructure
+//   This file is distributed under the University of Illinois Open Source
+//   License. See LICENSE.TXT for details.
+// -------------------------------------------------------------------------
+
+include "shared.thrift"
+
+namespace py ProductManagement
+namespace js codeCheckerProductManagement
+
+
+/*
+struct PrivilegeRecord {
+  1: string   name,
+  2: bool     isGroup
+}
+typedef list<PrivilegeRecord> PrivilegeRecords
+*/
+
+struct DatabaseConnection {
+  1:          string engine,         // The database engine, such as "sqlite" or "postgresql".
+  2:          string host,
+  3:          i32    port,
+  4:          string username_b64,
+  5: optional string password_b64,   // Database password is NOT sent server->client!
+  6:          string database
+}
+
+/* ProductConfiguration carries administrative data regarding product settings. */
+struct ProductConfiguration {
+  1:          i64                id,
+  2:          string             endpoint,
+  3:          string             displayedName_b64,
+  4:          string             description_b64,
+  5: optional DatabaseConnection connection
+}
+typedef list<ProductConfiguration> ProductConfigurations
+
+/* Product carries data to the end user's product list and tasks. */
+struct Product {
```

Comment:
For **each** product, this would mean the following:

 * Construction of a scope-specific permission configuration with the individual product's ID
 * Encoding this data and making an API call
 * The server opening a session on the configuration database and querying whether the user has rights to view the product
    * This might incur more than one query due to permission inheritance
 * Creating the response and sending it back
 * The UI handling this

For **each** product. Which we don't know how many will be.

This will be almost as bad if for **each** `Run` we would individually query the number of reports **for each** detection status. AKA:

 * getDetectionStatusForRun(run_id, NEW)
 * gDSFR(id, UNRESOLVED)
 * ...

For **each** run.

Whereas this data (especially whether or not the database connection to the product could be made) is readily available at product listing, and is an **integral part** of getting user-facing information about the product.

Reply:
@gyorb You yourself had to make #844 to speed up the queries. Your comment here states the exact opposite of what you seemed to *have to do* in #844 to make the system faster.

Reply:
Yes I understand this, but I still think clean api is better which does/checks only one thing. 
I did not sad we have to change it I've just sad I prefer not to mix multiple types of checks
"product endpoint exists" and "the user has access rights to it" into one api call.
On my query speed up I did not change the behavior of the api call or merged together with another just to perform better. I've improved an already existing api call which does only one thing.